### PR TITLE
fix Link to 'View' results

### PR DESCRIPTION
**Pull request by efvink**
1. fix Link to 'View' results 
2. don't show this view if there are no results

Screenshot:
image

image

### CODE REVIEW

File: src/SDK/MonitoringApplianceEmulator/MainWindow.xaml [hunk 1]
```diff
@@ -78,12 +79,7 @@
                 </Style>
             </TabControl.Resources>
 
-            <TabItem IsEnabled="False">
-                <TabItem.Header>
-                    <Separator Visibility="Hidden" Height="5"/>
-                </TabItem.Header>
-            </TabItem>
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/MainWindow.xaml]: why did you remove this? screenshot please

File: src/SDK/MonitoringApplianceEmulator/Controls/EmulationStatusControl.xaml [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
+    <UserControl.Resources>
+        <converters:CountToVisibilityConverter x:Key="CountToVisibilityConverter"/>
+    </UserControl.Resources>
     <StackPanel Orientation="Vertical">
         <TextBlock Style="{StaticResource ControlText}" Text="Alerts found" Margin="0,0,0,10"/>
-        <TextBlock Margin="0,0,0,20" HorizontalAlignment="Left">
-            <Run FontSize="30" Text="{Binding Path=SmartDetectorRunner.Alerts.Count, Mode=OneWay}"/>
-            <Separator Width="5" Visibility="Hidden"/>
-            <Hyperlink NavigateUri="http://stackoverflow.com">View</Hyperlink>
+        <TextBlock Margin="0,0,0,20" HorizontalAlignment="Left"><Run FontSize="30" Text="{Binding SmartDetectorRunner.Alerts.Count, Mode=OneWay}"/>
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/EmulationStatusControl.xaml]: >< [](start = 64, length = 1)

move to next line

File: src/SDK/MonitoringApplianceEmulator/Converters/EnumToIntConverter.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        /// <returns>
+        /// A converted value.
+        /// </returns>
+        /// <param name="value">The value that is produced by the binding target.</param>
+        /// <param name="targetType">The type to convert to.</param>
+        /// <param name="parameter">The converter parameter to use.</param>
+        /// <param name="culture">The culture to use in the converter.</param>
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return (Enum)Enum.Parse((Type)targetType, value.ToString());
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Converters/EnumToIntConverter.cs]: >value.ToString() [](start = 54, length = 16)

that won't work - you need to do a hard cast as well

File: src/SDK/MonitoringApplianceEmulator/ViewModels/EmulationStatusControlViewModel.cs [hunk 4]
```diff
@@ -20,6 +20,8 @@ public class EmulationStatusControlViewModel : ObservableObject
 
         private ITracer tracer;
 
+        private INotificationService notificationService;
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/EmulationStatusControlViewModel.cs]: make all of these readonly please

File: src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs [hunk 5]
```diff
@@ -10,6 +10,22 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringAppliancEmulator.V
...
+        /// <summary>
+        /// The Smart Detector Configuration Control
+        /// </summary>
+        SmartDetectorConfigurationControl,
+
+        /// <summary>
+        /// The Alerts Control
+        /// </summary>
+        AlertsControl
+    }
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs]: move to a separate file

File: src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs [hunk 6]
```diff
@@ -18,6 +34,7 @@ public class MainWindowViewModel : ObservableObject
         private int numberOfAlertsFound;
         private SmartDetectorRunner smartDetectorRunner;
         private string userName;
+        private TabItem selectedTab = TabItem.SmartDetectorConfigurationControl;
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs]: readonly

File: src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs [hunk 7]
```diff
@@ -34,15 +51,17 @@ public MainWindowViewModel()
...
-        public MainWindowViewModel(AlertsRepository alertsRepository, AuthenticationServices authenticationServices, SmartDetectorRunner smartDetectorRunner)
+        public MainWindowViewModel(AlertsRepository alertsRepository, AuthenticationServices authenticationServices, SmartDetectorRunner smartDetectorRunner, INotificationService notificationService)
         {
             this.NumberOfAlertssFound = 0;
             alertsRepository.Alerts.CollectionChanged +=
                 (sender, args) => { this.NumberOfAlertssFound = args.NewItems.Count; };
 
             this.UserName = authenticationServices.AuthenticationResult.UserInfo.GivenName;
             this.SmartDetectorRunner = smartDetectorRunner;
+            notificationService.SwitchTabEvent += () => { this.SelectedTab = TabItem.AlertsControl; };
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs]: >notificationService.SwitchTabEvent += () => { this.SelectedTab = TabItem.AlertsControl; }; [](start = 12, length = 90)

so this is wrong
either you:
a. name the event SwitchTabToAlertsControl, or
b. send the name of the new tab in an event args

I vote for b.

File: src/SDK/MonitoringApplianceEmulator/Models/INotificationService.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    /// <summary>
+    /// Interface providing notification service. Can be used to Invoke and Subscribe to events.
+    /// </summary>
+    public interface INotificationService
+    {
+        /// <summary>
+        /// Occurs when the user switch tabs from nested view
+        /// </summary>
+        event SwitchTabEventHandler SwitchTabEvent;
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Models/INotificationService.cs]: >SwitchTabEvent [](start = 36, length = 14)

don't put the type (event) in the property name. should be named TabSwitched

File: src/SDK/MonitoringApplianceEmulator/Models/INotificationService.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+        /// <summary>
+        /// Helper method to invoke switching tabs request
+        /// </summary>
+        void OnSwitchTabEventHandler();
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Models/INotificationService.cs]: >OnSwitchTabEventHandler [](start = 13, length = 23)

same naming reasoning - OnTabSwitched

File: src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs (hunk 7 above)
Comment (EranGon) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs]: actually, both b and some sort of a - the notification service is not used for some generic notification on some generic tab switch. the event that was fired represents that the "user has requested to view the alerts" - this is the name you should use

---
In reply to: [186691438](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186691438) [](ancestors = 186691438)

File: src/SDK/MonitoringApplianceEmulator/MainWindow.xaml (hunk 1 above)
Comment (efvink) [reply src/SDK/MonitoringApplianceEmulator/MainWindow.xaml]: You can see the screenshot in the PR. I used Margin instead because I wanted the indexes to match the enum)

---
In reply to: [186690520](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186690520) [](ancestors = 186690520)

File: src/SDK/MonitoringApplianceEmulator/ViewModels/EmulationStatusControlViewModel.cs (hunk 4 above)
Comment (efvink) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/EmulationStatusControlViewModel.cs]: All of these?

---
In reply to: [186690911](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186690911) [](ancestors = 186690911)

File: src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs (hunk 7 above)
Comment (efvink) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs]: What do you think about 'TabSwitchedToAlertsControl' and 'OnTabSwitchedToAlertsControl'? 

---
In reply to: [186692254](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186692254) [](ancestors = 186692254,186691438)

File: src/SDK/MonitoringApplianceEmulator/Converters/EnumToIntConverter.cs (hunk 3 above)
Comment (efvink) [reply src/SDK/MonitoringApplianceEmulator/Converters/EnumToIntConverter.cs]: It works with numeric value as string :)
Parse - "Converts the string representation of the name or numeric value of one or more enumerated constants to an equivalent enumerated object"

---
In reply to: [186690823](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186690823) [](ancestors = 186690823)

File: src/SDK/MonitoringApplianceEmulator/ViewModels/EmulationStatusControlViewModel.cs (hunk 4 above)
Comment (EranGon) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/EmulationStatusControlViewModel.cs]: yes - they are only changed in the ctor

---
In reply to: [186753580](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186753580) [](ancestors = 186753580,186690911)

File: src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs (hunk 7 above)
Comment (EranGon) [reply src/SDK/MonitoringApplianceEmulator/ViewModels/MainWindowViewModel.cs]: good choice!

---
In reply to: [186756030](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186756030) [](ancestors = 186756030,186692254,186691438)

File: src/SDK/MonitoringApplianceEmulator/Converters/EnumToIntConverter.cs (hunk 3 above)
Comment (EranGon) [reply src/SDK/MonitoringApplianceEmulator/Converters/EnumToIntConverter.cs]: yes, but still - no point in changing int to a string and then parsing it to an enum. a simple cast is way more efficient

---
In reply to: [186768927](https://github.com/liadtal/Smart-Alerts/pull/147#discussion_r186768927) [](ancestors = 186768927,186690823)
